File: 0016-3sum-closest/0016-3sum-closest.py
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        n = len(nums)
        nums.sort()
        closest = nums[0]+nums[0+1]+nums[n-1]
        for i in range(n):
            left = i+1
            right = n-1
            while left < right:
                sum = nums[i] + nums[left] + nums[right]
                if abs(sum - target) < abs(closest - target):
                    closest = sum
                if sum < target:
                    left += 1
                else:
                    right -= 1
        return closest

File: 0016-3sum-closest/0016-3sum-closest.cpp
class Solution {
public:
    int threeSumClosest(vector<int>& nums, int target) {
        int n = nums.size();
        sort(nums.begin(),nums.end());
        int closest = nums[0]+nums[0+1]+nums[n-1];
        for(int i= 0;i<n;i++){
            int l = i+1;
            int r = n-1;
            while(l<r){
                int sum = nums[i] + nums[l] + nums[r];
                if(abs(sum-target) < abs(closest - target)) {
                    closest = sum;
                }
                if(sum < target) l++;
                else r--;

            }
        }
        return closest;
    }
};